examples/ConstraintIB/eel2d/example.cpp

```diff
@@ -238,6 +240,40 @@ run_example(int argc, char* argv[])
         ib_method_ops->registerConstraintIBKinematics(ibkinematics_ops_vec);
         ib_method_ops->initializeHierarchyOperatorsandData();
 
+        // Create hydrodynamic force evaluator object.
+        double rho_fluid = input_db->getDouble("RHO");
+        double mu_fluid = input_db->getDouble("MU");
+        double start_time = time_integrator->getIntegratorTime();
+        Pointer<IBHydrodynamicForceEvaluator> hydro_force =
+            new IBHydrodynamicForceEvaluator("IBHydrodynamicForce", rho_fluid, mu_fluid, start_time, true);
+
+        // Get the initial box position and velocity from input
+        const string init_hydro_force_box_db_name = "InitHydroForceBox_0";
+        SAMRAI::tbox::Array<double> box_X_lower_array, box_X_upper_array, box_init_vel_array;
+        IBTK::Vector3d box_X_lower, box_X_upper, box_init_vel;
+
+        box_X_lower_array = input_db->getDatabase(init_hydro_force_box_db_name)->getDoubleArray("lower_left_corner");
+        box_X_upper_array = input_db->getDatabase(init_hydro_force_box_db_name)->getDoubleArray("upper_right_corner");
+        box_init_vel_array = input_db->getDatabase(init_hydro_force_box_db_name)->getDoubleArray("init_velocity");
+        for (int d = 0; d < 3; ++d)
+        {
+            box_X_lower[d] = box_X_lower_array[d];
+            box_X_upper[d] = box_X_upper_array[d];
+            box_init_vel[d] = box_init_vel_array[d];
+        }
+        hydro_force->registerStructure(box_X_lower, box_X_upper, patch_hierarchy, box_init_vel, 0);
+
+        // Create COM variable
+        std::vector<std::vector<double> > structure_COM = ib_method_ops->getStructureCOM();
```

Comment:
There are a number of calls where it looks like data are basically being copied directly from the IBMethod implementation to the force evaluator.

Instead of passing the structure COM, structure linear/angular momentum, etc, could the force evaluator class just take a pointer to the IBMethod implementation, and ask it for these values as needed? Seems less error prone (and certainly less user lines of code).

On the other hand: Are there cases where you don't want to get these values directly from the IBMethod implementation?